unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -850,7 +850,7 @@ public int compareTo(@Nonnull final UTF8String other) {
         return res;
       }
     }
-    return numBytes - other.numBytes;
+    return numBytes < other.numBytes ? -1 : (numBytes > other.numBytes ? 1 : 0);
```

Comment:
Oops, this is not something to change, since `numBytes` is nonnegative. Overflow in the subtraction is the usual reason for this fix, not readability. But this can't overflow.